src/services/dynamo-database-service.ts

```diff
@@ -148,4 +115,40 @@ export class DynamoDatabaseService {
       logger.info(`${LOGS_PREFIX_SENSITIVE_INFO} No intervention exists for this account ${userId}.`);
     }
   }
+
+  /**
+   * Function to configure query parameters for DynamoDB
+   * @param userId - userId from the relevant event
+   * @returns - the parameters needed to query DynamoDB
+   */
+  private inputParametersForDatabaseQuery(userId: string): QueryCommandInput {
+    logger.debug(`${LOGS_PREFIX_SENSITIVE_INFO} Attempting request to dynamo db, with ID : ${userId}`);
+    const parameters: QueryCommandInput = {
+      TableName: this.tableName,
+      KeyConditionExpression: '#pk = :id_value',
+      ExpressionAttributeNames: { '#pk': 'pk' },
+      ExpressionAttributeValues: { ':id_value': { S: userId } },
+    };
+    return parameters;
+  }
+
+  /**
+   * Function to validate the response and send metrics if there is a problem with the received response
+   * @param response - the response from dynamoDB
+   */
+  private validateQueryResponse(response: QueryCommandOutput): void {
```

Comment:
you could make this function return the first unmarshalled item in the array, if validation passes, so you do not need to do the `if (response.Items)` check in the caller method. `retrieveRecordsByUserId` for example you can call this, and then just return the result `as DynamoDBStateResult`